### DISCUSSION TITLE
mooseWarnings throw when Moose::_throw_on_error=true

### DISCUSSION THREAD

**Discussion by helen-brooks** [idaholab/moose#16720] [Q&A General] [answered]
Hi, 

It might be the intended behaviour, but it is currently the case that mooseWarnings throw when Moose::_throw_on_error=true, even if Moose::_warnings_are_errors = false. It would be nice to have the feature to control whether mooseErrors and mooseWarnings throw independently, either by checking both Moose::_throw_on_error and Moose::_warnings_are_errors inside 
mooseWarningStream or alternatively add an additional flag such as Moose::_throw_on_warning.
I wouldn't call this an issue, hence I'm putting this in Discussion, but it'd be simple to implement and for me at least it would be a helpful addition for writing certain unit tests.

**Comment by GiudGiud** [comment, marked as answer]
Thanks for the info. Seems like an issue to me, adding it to issues.
